python/lsst/pipe/analysis/visitAnalysis.py

```diff
@@ -255,22 +273,61 @@ def runDataRef(self, dataRefList, tract=None, subdir=""):
             if not dataRefListTract:
                 self.log.info("No data found for tract: {:d}".format(tractList[i]))
                 continue
-            repoInfo = getRepoInfo(dataRefListTract[0], doApplyUberCal=self.config.doApplyUberCal)
+            repoInfo = getRepoInfo(dataRefListTract[0],
+                                   doApplyExternalPhotoCalib=self.config.doApplyExternalPhotoCalib,
+                                   externalPhotoCalibName=self.config.externalPhotoCalibName,
+                                   doApplyExternalSkyWcs=self.config.doApplyExternalSkyWcs,
+                                   externalSkyWcsName=self.config.externalSkyWcsName)
             self.log.info("dataId: {!s:s}".format(repoInfo.dataId))
-            ccdListPerTract = getDataExistsRefList(dataRefListTract, repoInfo.dataset)
-            if not ccdListPerTract and self.config.doApplyUberCal:
-                # Check for wcs for compatibility with old dataset naming
-                ccdListPerTract = getDataExistsRefList(dataRefListTract, "wcs")
-                if ccdListPerTract:
-                    self.log.info("Old meas_mosaic dataset naming: wcs (new name is jointcal_wcs)")
-                    repoInfo.dataset = "wcs"
-            self.log.info("Exising data for tract {:d}: ccdListPerTract = {}".
-                          format(tractList[i], ccdListPerTract))
+            ccdListPerTract = getDataExistsRefList(dataRefListTract, repoInfo.catDataset)
             if not ccdListPerTract:
-                if self.config.doApplyUberCal:
-                    self.log.fatal("No data found for {:s} datset...are you sure you ran meas_mosaic? "
-                                   "If not, run with --config doApplyUberCal=False".format(repoInfo.dataset))
-                raise RuntimeError("No datasets found for datasetType = {:s}".format(repoInfo.dataset))
+                raise RuntimeError("No datasets found for datasetType = {:s}".format(repoInfo.catDataset))
+            if self.config.doApplyExternalPhotoCalib:
+                ccdPhotoCalibListPerTract = getDataExistsRefList(dataRefListTract, repoInfo.photoCalibDataset)
+                if not ccdPhotoCalibListPerTract:
+                    self.log.fatal("No data found for {} dataset...are you sure you ran the external "
+                                   "photometric calibration?  If not, run with "
+                                   "--config doApplyExternalPhotoCalib=False".
+                                   format(repoInfo.photoCalibDataset))
```

Comment:
These and the below strings might be good candidates for the f-strings, e.g.:
```
self.log.fatal(f"No data found for {repoInfo.photoCalibDataset}...are you sure you ran the external "
               "photometric calibration?  If not, run with "
               "--config doApplyExternalPhotoCalib=False")
```

Reply:
I replaced most occurrences touched here, but will leave a full conversion (if that is what our standards would prefer) to a later ticket (likely DM-14559).